.vscode/settings.json

```diff
@@ -2,14 +2,11 @@
   "python.analysis.extraPaths": [".virtualenv"],
   "[python]": {
     "editor.formatOnSave": true,
-    "editor.defaultFormatter": "ms-python.black-formatter",
+    "editor.defaultFormatter": "charliermarsh.ruff",
     "editor.rulers": [88],
     "editor.codeActionsOnSave": {
+      "source.fixAll": "explicit",
```

Comment:
@eliotwrobson I'm guessing this enables Ruff to fix sort order on file save? Per https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff:

> You can configure Ruff to fix lint violations on-save by enabling the source.fixAll action in settings.json:
> ```json
> {
>   "[python]": {
>     "editor.codeActionsOnSave": {
>       "source.fixAll": "explicit"
>     }
>   }
> }
> ```

I'm surprised this is necessary since the [isort VS Code extension](https://marketplace.visualstudio.com/items?itemName=ms-python.isort) was able to update files automatically without the need for additional configuration.

Reply:
Ahh this is because the ruff formatter and ruff linter are slightly different. Adding this enable fixing lint violations on save, but that's different than the formatter itself (I think).

Reply:
@eliotwrobson Fair enough. I would think that formatting violations technically constitute linting violations (sort of how they do in flake8 and flake8-black, e.g. the "Black would require changes" linter message).

Let's keep it, though! We want to have compliant and quality code in the codebase, and enforcing it through automation (i.e. `source.fixAll`) helps maintain that standard.